File: backend/src/main/java/com/apocalypse/thefall/repository/CharacterTileDiscoveryRepository.java
package com.apocalypse.thefall.repository;

import com.apocalypse.thefall.entity.CharacterTileDiscovery;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface CharacterTileDiscoveryRepository extends JpaRepository<CharacterTileDiscovery, Long> {

    List<CharacterTileDiscovery> findByCharacterId(Long characterId);
}

File: backend/src/main/java/com/apocalypse/thefall/repository/CharacterTileDiscoveryRepository.kt
package com.apocalypse.thefall.repository

import com.apocalypse.thefall.entity.CharacterTileDiscovery
import org.springframework.data.jpa.repository.JpaRepository

interface CharacterTileDiscoveryRepository : JpaRepository<CharacterTileDiscovery, Long> {
    fun findByCharacterId(characterId: Long): List<CharacterTileDiscovery>
}
